### ISSUE TITLE
Account linking

### ISSUE THREAD

**Issue by CiriousJoker** [funwithflutter/lit_firebase_auth#16]
It would be nice if there was some sort of account linking.

An extra function would be nice for each provider, meaning instead of `contrext.signInWithGoogle()`you call `context.linkToGoogle()`;

This could automatically be switched if the user is already logged in.

This feature enables anonymous users to link their account on multiple devices. It would be quite helpful.

**Comment by funwithflutter** [comment]
Yeah, for sure! I'll schedule some time in the next few days to catch up on all of the outstanding feature requests 😁. But feel free to make a pull request if you don't mind jumping into the code

**Comment by percula** [comment]
Hi @funwithflutter , have you started on this yet? If not, I'll give it a shot. I need it for my app release in ~1 month.
